### DISCUSSION TITLE
improve helm chart

### DISCUSSION THREAD

**Discussion by szaimen** [nextcloud/all-in-one#3742] [Ideas] [answered]
- [x] make mail config configurable via helm chart https://github.com/nextcloud/all-in-one/pull/3756
- [x] make app include list configurable via helm chart https://github.com/nextcloud/all-in-one/pull/3756
- [x] allow to add subscription key via env https://github.com/nextcloud/all-in-one/pull/3756
- [x] ~~make storage customization also configurable via helm chart~~ not sure what is meant here but was decided against
- [x] ~~load balancer possibility to start Apache first (check +feedback if feasible); maybe use https://github.com/nextcloud/all-in-one/discussions/3576?~~ [tried but not feasible](https://github.com/nextcloud/all-in-one/discussions/3742#discussioncomment-7577890) -> implement logic to get ip-address and adjust firewall config automatically
- [x] ~~nodes do not come up again, currently no re-scheduling (check, possibly longhorn config topic); check if using ReadWriteMany fixes this~~ recommended to set some longhorn settings differently...
- [x] ~~Pin included apps to specific versions?~~ Hopefully not necessary due to https://github.com/nextcloud/all-in-one/pull/3707
- [x] ~~apache does not restart together with nextcloud and does not get its new ip-address on new deploy...~~ tested on our own and seems to work

**Comment by st3iny** [comment]
I developed a unified load balancer POC and thoroughly discussed it with Simon. Aside from the fact that there were problems with Talk, we came to the conclusion to drop this idea.

Maintaining a dedicated load balancer container just for Helm is not feasible as we won't be able to guarantee stability. It won't be a part of the regular dockerized AIO setup and thus not much testing would be done on it. 

Additionally, it might be tedious to maintain as my implementation turned out to be very brittle and many things had to be fixed by trial and error. 

So the recommended solution from our side is indeed that the customer implements a way to retreive the ip-address that got allocated by kubernetes and autommatically adjust the firewall rules based on that. (Could potentially be done via a script using kubectl).

**Comment by szaimen** [comment, marked as answer]
Most things get added in https://github.com/nextcloud/all-in-one/pull/3756
